draft-ietf-jose-json-web-proof.md

```diff
@@ -102,6 +102,248 @@ While `issue` and `confirm` only occur when a JWP is initially created by the is
+the kind of object is already known.
+This parameter is ignored by JWP implementations;
+any processing of this parameter is performed by the JWP application.
+Use of this Header Parameter is OPTIONAL.
+
+Per [@RFC2045], all media type values,
+subtype values, and parameter names are case insensitive.
+However, parameter values are case sensitive unless
+otherwise specified for the specific parameter.
+
+To keep messages compact in common situations,
+it is RECOMMENDED that producers omit an "application/" prefix
+of a media type value in a `typ`
+Header Parameter when no other '/' appears in the media type value.
+A recipient using the media type value MUST treat it as if
+"application/" were prepended to any `typ` value not containing a '/'.
+For instance, a `typ` value of `example` SHOULD be used to represent
+the `application/example` media type,
+whereas the media type `application/example;part="1/2"` cannot
+be shortened to `example;part="1/2"`.
+
+The `typ` value `jwp` can be used by applications
+to indicate that this object is a JWP using the JWP Compact Serialization.
+The `typ` value `jwp+json` can be used by applications
+to indicate that this object is a JWP using the JWP JSON Serialization.
+Other type values can also be used by applications.
+
+It is RECOMMENDED that the `typ` Header Parameter be used for explicit typing,
+in parallel to the recommendations in Section 3.11 of [@RFC8725].
+
+<!--- It's not clear that `cty` makes sense for JWPs; retaining text here for now.
+
+### "cty" (Content Type) Header Parameter {#ctyDef}
+
+The `cty` (content type) Header Parameter
+is used by JWP applications to declare the
+media type (#IANA.MediaTypes)
+of the secured content (the payload).
+This is intended for use by the application when
+more than one kind of object could be present in
+the JWP Payload;
+the application can use this value to disambiguate among
+the different kinds of objects that might be present.
+It will typically not be used by applications when
+the kind of object is already known.
+This parameter is ignored by JWP implementations;
+any processing of this parameter is performed by the JWP application.
+Use of this Header Parameter is OPTIONAL.
+
+Per [@RFC2045], all media type values,
+subtype values, and parameter names are case insensitive.
+However, parameter values are case sensitive unless
+otherwise specified for the specific parameter.
+
+To keep messages compact in common situations,
+it is RECOMMENDED that producers omit an "application/" prefix
+of a media type value in a `cty`
+Header Parameter when no other '/' appears in the media type value.
+A recipient using the media type value MUST treat it as if
+"application/" were prepended to any
+`cty` value not containing a '/'.
+For instance, a `cty` value of
+`example` SHOULD be used to represent
+the `application/example` media type,
+whereas the media type
+`application/example;part="1/2"` cannot
+be shortened to `example;part="1/2"`.
+
+-->
+
```

Comment:
We can go ahead and remove `cty`; we instead would establish a 'layout' that describes each payload slot's meaning.

Reply:
```suggestion
```